### PR TITLE
move string.getObject to its own module 

### LINKED ISSUE #110: move string.getObject
In the `can-util/js/string/string` module, there is a `getObject` method: http://canjs.github.io/canjs/doc/can-util/js/string/string.getObject.html

Its API should be more like:

https://lodash.com/docs/4.16.2#get
- [x] - move into its own module, maybe tweak its API.
- [ ] - have string.getObject call this and still support multiple context ... but put a warning in string.getObject. We probably don't need to support `add` argument anymore.

### PR DESCRIPTION
**Pull request by sarathnair**
Closes #110

### CODE REVIEW

File: js/get-object/get-object.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+var isArray = require('../is-array/is-array');
```
Comment (justinbmeyer) [inline js/get-object/get-object.js]: The module should probably be `js/get/get`.

File: js/get-object/get-object.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * if it's an array, `roots` itself otherwise.
+ * 
+ * ```js
+ * var getObject = require("can-util/js/getObject/getObject");
+ * console.log(getObject("a.b.c", {a: {b: {c: "foo"}}})); // -> "foo"
+ * console.log(getObject("a.b.c", {a: {}})); // -> undefined
+ * console.log(getObject("a.b", [{a: {}}, {a: {b: "bar"}}])); // -> "bar"
+ * ```
+ */
+function getObject(name, roots) {
```
Comment (justinbmeyer) [inline js/get-object/get-object.js]: do we need to support multiple roots?  Could can-control be fixed to avoid this?

**Review by justinbmeyer — DISMISSED** [review]

File: js/get-object/get-object.js (hunk 1 above)
Comment (sarathnair) [reply js/get-object/get-object.js]: Ok I can make that change

File: js/get-object/get-object.js (hunk 2 above)
Comment (sarathnair) [reply js/get-object/get-object.js]: I am not sure about that Justin, supporting multiple roots looked kind of cool to me. Let me know your views

**Comment by daffl** [comment]
Thank you @sarathnair! I fixed the tests and updated the signature and documentation. Will merge and make a 3.0 release.

**Pull request merged**